### PR TITLE
Print virtual address already mapped in user error

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
Useful for debugging in my experience.

### CODE REVIEW

**Review by lsf37 — APPROVED** [review]

**Comment by lsf37** [comment]
I approved a bit too quickly: yes, it is useful and we should add it, but are there other architectures that throw the same error or do they all allow over-mapping? If there are others, we should add it there as well.

**Comment by Ivan-Velickovic** [comment]
> do they all allow over-mapping?

I thought they all allow over-mapping. I did a quick grep and could not see any other similar messages for pages themselves.

**Comment by Ivan-Velickovic** [comment]
Yea from https://github.com/seL4/seL4/pull/967 it looks like only RISC-V prevents over-mapping.

**Comment by lsf37** [comment]
> Yea from #967 it looks like only RISC-V prevents over-mapping.

You're right, good to merge after all.

File: src/arch/riscv/kernel/vspace.c [hunk 1]
```diff
@@ -888,7 +888,7 @@ static exception_t decodeRISCVFrameInvocation(word_t label, word_t length,
         } else {
             /* check this vaddr isn't already mapped */
             if (unlikely(pte_ptr_get_valid(lu_ret.ptSlot))) {
-                userError("Virtual address already mapped");
+                userError("Virtual address (0x%lx) already mapped", vaddr);
```
Comment (axel-h) [inline src/arch/riscv/kernel/vspace.c:891]:
```suggestion
                userError("Virtual address (0x%"SEL4_PRIx_word") already mapped", vaddr);
```

**Pull request merged**